### DISCUSSION TITLE
I've never seen this typing syntax. What is it?

### DISCUSSION THREAD

**Discussion by qexat** [microsoft/pylance-release#3292] [Q&A] [answered]
I was coding some random stuff when I discovered a typing syntax I've never seen before:
![image](https://user-images.githubusercontent.com/43090614/188709222-c073865b-4eab-409a-b56b-0d03cc87230d.png)
I suppose it is related to `TypeVar` since the parameter is one until I do some `if isinstance()` checks then it falls back to this `object*`, but what does the star suffix mean?

**Comment by erictraut** [comment, marked as answer]
See [this documentation](https://github.com/microsoft/pyright/blob/main/docs/type-concepts.md#constrained-type-variables-and-conditional-types) for details. It effectively means that the type is an intersection between a concrete type and a type variable. In other words, it's an `object` but it's also a `T` (or whatever TypeVar it corresponds to).

**Reply by echoix** [reply]
For future reference, I think that the section referred to is now in this document https://github.com/microsoft/pyright/blob/8f35aa06437f517a03fc68a4ae086ede4e1b6f83/docs/type-concepts-advanced.md#conditional-types-and-type-variables (permalink)
